rd-cpp/src/rd_framework_cpp/src/main/protocol/Buffer.cpp

```diff
@@ -190,7 +205,7 @@ int64_t TICKS_PER_MILLISECOND = 10000000;
 DateTime Buffer::read_date_time()
 {
 	int64_t time_in_ticks = read_integral<int64_t>();
-	time_t t = static_cast<time_t>(time_in_ticks / TICKS_PER_MILLISECOND - TICKS_AT_EPOCH / TICKS_PER_MILLISECOND);
+	time_t t = static_cast<time_t>((time_in_ticks - TICKS_AT_EPOCH) / TICKS_PER_MILLISECOND);
```

Comment:
Here, you're rolling back a change purposefully made at https://github.com/JetBrains/rd/commit/4be1d02063cfaff0781b7ae23a1822142505a8ee.

Are you sure this should be done? Perhaps we should fix `write_date_time` instead, too?

Reply:
Sorry, I think I messed up when ported changes from UnrealLink. I though that these were made there, and no changes were in rd-cpp repo. I will reevaluate changes

Reply:
I think I've marked every case when something wrong was happening. Everything else LGTM.

Reply:
Thank you!